bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/client/ConsumedThingImpl.java

```diff
@@ -0,0 +1,246 @@
+            for (var link : propertyDescription.links) {
+                if ((link.rel != null) && (link.href != null) && link.rel.equals("property")) {
+                    return webThingURI.resolve(link.href);
+                }
+            }
+        }
+        throw new RuntimeException(
+                "WebThing " + webThingURI + " does not support a property uri. WebThing description: " + description);
+    }
+
+    private URI getEventStreamUri() {
+        for (var link : this.description.links) {
+            var href = link.href;
+            if (href != null) {
+                var rel = Optional.ofNullable(link.rel).orElse("<undefined>");
+                if (rel.equals("alternate")) {
+                    return URI.create(href);
+                }
+            }
+        }
+        throw new RuntimeException("WebThing " + webThingURI + " does not support websocket uri. WebThing description: "
+                + this.description);
+    }
+
+    @Override
+    public boolean isAlive() {
+        return isOpen.get() && this.websocketDownstream.isAlive();
+    }
+
+    @Override
+    public void close() {
+        logger.debug("WebThing {} closed called", webThingURI);
+        isOpen.set(false);
+        this.websocketDownstream.close();
+    }
+
+    void onError(String reason) {
+        logger.debug("WebThing {} error occurred. {}", webThingURI, reason);
+        if (isOpen.get()) {
+            errorHandler.accept(reason);
+        }
+        close();
+    }
+
+    @Override
+    public WebThingDescription getThingDescription() {
+        return this.description;
+    }
+
+    @Override
+    public void observeProperty(String propertyName, BiConsumer<String, Object> listener) {
+        this.websocketDownstream.observeProperty(propertyName, listener);
+
+        // it may take a long time before the observed property value will be changed. For this reason
+        // read and notify the current property value (as starting point)
+        try {
+            var value = readProperty(propertyName);
+            listener.accept(propertyName, value);
+        } catch (PropertyAccessException pae) {
+            logger.warn("could not read WebThing {} property {}", webThingURI, propertyName, pae);
+        }
+    }
+
+    @Override
+    public Object readProperty(String propertyName) throws PropertyAccessException {
+        var propertyUri = getPropertyUri(propertyName);
+        try {
+            var response = httpClient.newRequest(propertyUri).timeout(30, TimeUnit.SECONDS).send();
+            if (response.getStatus() < 200 || response.getStatus() >= 300) {
+                onError("WebThing " + webThingURI + " disconnected");
+                throw new PropertyAccessException("could not read " + propertyName + " (" + propertyUri + ")");
+            }
+            var body = response.getContentAsString();
+            var properties = new Gson().fromJson(body, Map.class);
```

Comment:
You could make the Gson object a final field to save resources. Same for below.

Reply:
fixed